solidity/contracts/SisypheanExchange.sol

```diff
@@ -1,21 +1,259 @@
 pragma solidity 0.8.30;
 
-import './Universe.sol';
+import './Reporting.sol';
+import './ForkedERC1155.sol';
+import './Constants.sol';
+import './ISisypheanExchange.sol';
+import './ReputationToken.sol';
+import './IERC20.sol';
 
-contract SisypheanExchange {
+// NOTE: Currently a resolved market could be traded on and resolved again in a child universe. We can add things to prevent this if needed.
+// NOTE: We need a mechanism for having certain Cash be 1:1 while other existing cash is potentially a different ratio. Consider someone putting 1 eth into a complete set in a market and it migrating while unresolved in a fork. Post auction lets say it hit 90% fulfilment. The original complete set should provide .9 ETH in return but a new purchaser should be able to buy a complete set worth 1 ETH.
```

Comment:
I don't think this comment is correct.  In a universe that suffers a 10% loss in CASH value, new complete sets would then cost 0.9 ETH to buy.  Essentially, all that changes in a fork with an underfilled auction is the face value of CASH changes from 1:1 to 1:<1.